applications/FluidDynamicsApplication/custom_elements/data_containers/derivatives.h

```diff
@@ -0,0 +1,172 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_DERIVATIVES_H)
+#define KRATOS_DERIVATIVES_H
+
+// System includes
+#include <tuple>
+#include <utility>
+
+// External includes
+
+// Project includes
+#include "includes/ublas_interface.h"
+
+// Application includes
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+template<unsigned int TNumNodes, unsigned int TBlockSize>
+class ZeroDerivatives
+{
+public:
+    ///@name Local type definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    constexpr static IndexType NumNodes = TNumNodes;
+
+    constexpr static IndexType BlockSize = TBlockSize;
+
+    ///@}
+    ///@name Public static operations
+    ///@{
+
+    template<IndexType TSize, class... TArgs>
+    void inline CalculateGaussPointResidualsDerivativeContributions(BoundedVector<double, TSize>& rResidualDerivative, TArgs&... rArgs) const
```

Comment:
```suggestion
    void inline CalculateGaussPointResidualsDerivativeContributions(
        BoundedVector<double, TSize>& rResidualDerivative,
        TArgs&... rArgs) const
```
Style guide comment (multiline for non-single argument methods/functions).